pkgs/build-support/fetchmavenartifact/default.nix

```diff
@@ -0,0 +1,57 @@
+# Adaptation of the MIT-licensed work on `sbt2nix` done by Charles O'Farrell
+
+{ fetchurl, stdenv }:
+let
+  defaultRepos = [
```

Comment:
I would move the list to the `pkgs/build-support/fetchurl/mirrors.nix` and then use something like `mirror://maven/` here. Then appending to the list isn't that hard:
```nix
fetchMavenArtifact {
  # ...
  repos = [ mirror://maven/ http:/mymirror.com/ ];
}
```

That also decouples policy from mechanism.